### PR TITLE
fix: Stop workers on interrupted event

### PR DESCRIPTION
**Pull request by ludeknovy**
🚨Please review the [guidelines for contributing](https://github.com/schemathesis/schemathesis/blob/master/CONTRIBUTING.rst) to this repository.

### Description
Fix for #1066

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Added a changelog entry
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -4,6 +4,12 @@ Changelog
 `Unreleased`_ - TBD
 -------------------
 
+**Fixed**
+- Worker threads may not be immediately stopped on SIGINT. `#1066`_
```
Comment (Stranger6667) [inline docs/changelog.rst:8]: Please, put this line under the existing `Fixed` subsection below

**Review by Stranger6667 — CHANGES_REQUESTED** [review]
Awesome! Thank you for fixing this :) I have a minor comment on the changelog entry

File: docs/changelog.rst (hunk 1 above)
Comment (Stranger6667) [reply docs/changelog.rst:8]: It also needs `.. _#1066: https://github.com/schemathesis/schemathesis/issues/1066` line below in this file (around the line 2140)

Comment (ludeknovy) [reply docs/changelog.rst]: Sorry, I messed it up... Pushed an update.

Comment (Stranger6667) [reply docs/changelog.rst]: That's perfectly fine :) thanks

**Review by Stranger6667 — APPROVED** [review]

**Comment by Stranger6667** [comment]
Thank you @ludeknovy :)

**Pull request merged**